clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonCloudFormationProviderSpec.groovy

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2019 Schibsted Media Group.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.aws.provider.view
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.cats.cache.DefaultCacheData
+import com.netflix.spinnaker.cats.mem.InMemoryCache
+import com.netflix.spinnaker.clouddriver.aws.cache.Keys
+import com.netflix.spinnaker.clouddriver.aws.model.AmazonCloudFormation
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+import static com.netflix.spinnaker.clouddriver.aws.cache.Keys.Namespace.CLOUDFORMATION
+
+class AmazonCloudFormationProviderSpec extends Specification {
+  static final TypeReference<Map<String, Object>> ATTRIBUTES = new TypeReference<Map<String, Object>>() {}
+
+  @Subject
+  AmazonCloudFormationProvider provider
+
+  ObjectMapper objectMapper = new ObjectMapper()
+
+  def setup() {
+    def cache = new InMemoryCache()
+    cloudFormations.each {
+      cache.merge(CLOUDFORMATION.ns,
+          new DefaultCacheData(makeKey(it), objectMapper.convertValue(it, ATTRIBUTES), [:]))
+    }
+
+    provider = new AmazonCloudFormationProvider(cache, objectMapper)
+  }
+
+  @Unroll
+  void "list all cloud formations by account (any region)"() {
+    when:
+    def result = provider.list(accountId, '*') as Set
+
+    then:
+    result == cloudFormations.findAll { it.accountId == accountId } as Set
+
+    where:
+    accountId  || count
+    "account1" || 2
+    "account2" || 1
+    "unknown"  || 0
+    null       || 0
```

Comment:
Are you wanting to assert anything with `count`?

Reply:
good question, it's a left over from a previous way of testing. I've changed the test a bit to make it more clear and explicit on the assertions.
done in https://github.com/spinnaker/clouddriver/pull/3278/commits/7455e0a53b9a12e6eed371e5cb7b9f4a895a7ec2